Develop1

adelikade opened the pull request.

adelikade committed:
First commit with tests

adelikade committed:
add jacoco report

adelikade committed:
Create README

nikolay-977 commented on pom.xml:
```diff
@@ -1,6 +1,6 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
```
Можно улучшить: в заготовке проекта есть лишние файлы, такие как test/java/tests/read.me. Их следует удалять.

adelikade replied:
Удалила

nikolay-977 commented on pom.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (mvn clean test). Как оформлять README.md можно ознакомиться в статье https://dzen.ru/a/YbzudflBPE1IDJF-

adelikade replied:
Изменила файл README.md в ветке develop1

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,19 @@
+    @Test
+    public void getPrice() {
+        bun = new Bun("red bun", 300);
+        Assert.assertEquals(300, bun.getPrice(), 0);
```
Необходимо исправить: не следует использовать числа, значение которых не является очевидным (антипатерн магические числа). Лучше сохранить в переменную наименованием, отражающим суть содержимого.

adelikade replied:
Внесла изменения в BunTest, IngredientFillingTest и IngredientSauceTest

nikolay-977 commented on src/test/java/IngredientFillingTest.java:
```diff
@@ -0,0 +1,24 @@
+import praktikum.Ingredient;
+import praktikum.IngredientType;
+
+public class IngredientFillingTest {
```
Необходимо исправить: Напиши тесты для класса IngredientType в классе IngredientTypeTest. Например, можно использовать метод IngredientType.valueOf("SAUCE").

adelikade replied:
Вынесла проверки на тип ингредиента в отдельный класс IngredientTypeTest

adelikade committed:
Update README.md

Добавление развернутой информации о проекте

adelikade committed:
Удалила лишний файл read.me
Добавила переменные с числами и строками в BunTest,IngredientFillingTest и IngredientSauceTest
Вынесла проверки на тип ингредиента в отдельный класс IngredientTypeTest

adelikade committed:
Merge remote-tracking branch 'origin/develop1' into develop1
